src/frontEnd/ProjectExplorer.py

```diff
@@ -226,51 +230,101 @@ def refreshProject(self):
         json.dump(self.obj_appconfig.project_explorer,
                   open(self.obj_appconfig.dictPath, 'w'))
 
-    '''def renameProject(self):
-        indexItem = self.treewidget.currentIndex()
-        baseFileName = str(indexItem.data())
-        newBaseFileName, ok = QtGui.QInputDialog.getText(
-            self, 'Rename Project', 'Project Name:',
-            QtGui.QLineEdit.Normal, baseFileName
-        )
-        if ok and newBaseFileName:
-            newBaseFileName = str(newBaseFileName)
-            projectPath, projectFiles = list(
-                self.obj_appconfig.project_explorer.items())[indexItem.row()]
+    def renameProject(self):
+        """
+        This function renames the project present in project explorer area
+        it validates first:
+        
+            - If project names is not empty.
+            - Project name does not contain spaces between them.
+            - Project name is different between what it was earlier.
+            - Project name should not exist.
+
+        And after project name is changed it recreates the project explorer tree.
+        """
+        self.indexItem = self.treewidget.currentIndex()
+        self.baseFileName = str(self.indexItem.data())
+        self.newBaseFileName, ok = QtGui.QInputDialog.getText(self, 'Rename Project', 'Project Name:',
```

Comment:
```suggestion
        newBaseFileName, ok = QtGui.QInputDialog.getText(self, 'Rename Project', 'Project Name:',
```
don't use self. for local variables